src/main/java/com/epam/esm/controller/GiftCertificateController.java

```diff
@@ -0,0 +1,70 @@
+package com.epam.esm.controller;
+
+import com.epam.esm.dao.DaoException;
+import com.epam.esm.model.Certificate;
+import com.epam.esm.model.Tag;
+import com.epam.esm.service.CertificateService;
+import com.epam.esm.service.CertificateServiceException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.util.MultiValueMap;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+
+@RestController
+@RequestMapping("/certificates")
+public class GiftCertificateController {
+    private CertificateService certificateService;
+
+    @Autowired
+    public GiftCertificateController(CertificateService certificateService) {
+        this.certificateService = certificateService;
+    }
+
+    @RequestMapping(method = RequestMethod.GET)
+    public List<Certificate> getByParameters(@RequestParam(required = false) MultiValueMap<String, String> params) throws DaoException {
+        try {
+            return certificateService.getByParameters(params);
+        } catch (CertificateServiceException e) {
+            throw new DaoException(e);
```

Comment:
Why did you wrap CertificateServiceException ?

Reply:
Why throws DaoException?

Reply:
Because here i want to catch my CertificateServiceException and throw DaoException